handlers.go

```diff
@@ -443,20 +446,78 @@ func isHistoryStale(history *ecrpc.PairData, threshold time.Duration) bool {
 }
 
 // mergePairData merges the pair data from two pairs based on the most recent
-// timestamp.
+// timestamp. It does the following:
+//   - It updates the success time and amounts if there are more recent history
+//     pairs, ensuring that the maximum success amount of the history pair is
+//     retained to prevent the success range from shrinking when unnecessary,
+//   - It prevents the failure from updating too soon based on the configured
+//     MinFailureRelaxInterval value.
+//   - It also adjusts the failure range if the success amount goes into the
+//     failure range and adjusts the success range if the failure amount goes
+//     into the success range.
+//
+// Parameters:
+// - existingData: The existing pair data to merge with.
+// - newData: The new pair data to merge with.
 func mergePairData(existingData, newData *ecrpc.PairData) {
-	// Update success time and amounts if the new data's success time is
-	// greater.
+
 	if newData.SuccessTime > existingData.SuccessTime {
+		// Update success time and amounts if newer, retaining max
+		// success amount to avoid shrinking success range
+		// unnecessarily.
 		existingData.SuccessTime = newData.SuccessTime
-		existingData.SuccessAmtSat = newData.SuccessAmtSat
-		existingData.SuccessAmtMsat = newData.SuccessAmtMsat
+		if newData.SuccessAmtMsat > existingData.SuccessAmtMsat {
+			existingData.SuccessAmtMsat = newData.SuccessAmtMsat
+		}
+
+		// Move the failure range up if the success amount goes into the
+		// failure range. We don't want to clear the failure completely
+		// because we haven't learnt much for amounts above the current
+		// success amount.
+		if existingData.FailTime != 0 &&
```

Comment:
I'm not sure I fully understand the question but this brings a problem to my mind regards this checking part one I just moved the adjust failure range up after updating the fail and success times/amounts otherwise this condition becomes stale in the success block since the fail amount may be updated but not reflected yet:
```
existingData.FailTime != 0 &&
		newData.SuccessAmtMsat >= existingData.FailAmtMsat
```

Reply:
Please let me know if that addresses the problem ?